Report Feature Done

mohammad-omar-01 opened the pull request:
Build the Third part of the REPORT feature

mohammad-omar-01 committed:
EDIT GIT IGNORE

mohammad-omar-01 committed:
REMOVE UNWANTED TRACKED FILES

mohammad-omar-01 committed:
DELETE UNWATNED FILES

mohammad-omar-01 committed:
BUILD REPORT FEATURE

mohammad-omar-01 committed:
GitIgnore Fix

mohammad-omar-01 committed:
UNTRACK UNWANTED DIRs

mohammad-omar-01 committed:
Solve Merge Conflict of .gitIgnoreFile

mohammad-omar-01 committed:
BUILD Selective DISCOUNT FEATURE

mohammad-omar-01 committed:
UPDATE PRODUCT PROPRITES

mohammad-omar-01 committed:
BUILD THE PRECENCE FEATURE

MazenAmria commented on Product.cs:
```diff
@@ -8,52 +8,54 @@ namespace Price_Calculator
+
         }
-        public Product(String name, int upc, double price, double flat_rate_task)
+        public Product(String name, int upc, double price, double flat_rate_tax, double discount)
```
here you're passing the tax rate and discount to each product individually.
Suppose we had like 100 products, and we wanted to change the tax rate from 20 to another value? how'd you solve it?

MazenAmria commented:
you should follow the naming convention of C#
and please make sure to do a code cleanup before pushing it, this helps formatting the code and removes redundant lines/data/commented code.

mohammad-omar-01 committed:
fix: Applying TAX to Multiple Product

Changes: Use separate class for tax and product and removed ProductUtilities class
which holds everything inside it, and i have introduced a new class called Report class with tasks related to the report
printing.
also includes a new class called StringFormatter to deal with formatting values in the required and right way.

mohammad-omar-01 committed:
Remove THE DISCOUNT class

mohammad-omar-01 committed:
Discount Built

mohammad-omar-01 committed:
BUILD DISCOUNT FEATURE
accoirding to your last comments on privous pull requests ive built this feature again using new technic
By building a new Discount Class with Idiscount interface, menu has been added to the system.

mohammad-omar-01 committed:
EDIT function name

mohammad-omar-01 committed:
Add No Discount Line

mohammad-omar-01 committed:
Edit Report

mohammad-omar-01 committed:
Other : updates of report feature hand enhacnments
